### ISSUE TITLE
Override IP on internal URL to prevent certificate issues

### ISSUE THREAD

**Issue by jschwalbe** [home-assistant/iOS#1621]
**Is your feature request related to a problem? Please describe.**

Currently the app has options for internal and external URLs. The problem with this is that the certificates don’t match up and there is an error and it won’t connect. 

**Describe the solution you'd like**
Add an option for something like “IP override” so that it still uses the same domain for verifying the certificate (ie duckdns, whatever), but will use the internal IP address for the connections. 

E.g.:
Assume something.duckdns.org resolves to an external IP. 
Internal URL: https://something.duckdns.org:8123 # required for the certificates to validate!
Internal IP override: 192.168.1.50
External URL: https://something.duckdns.org:9000

**Additional context**
Basically it’d be like editing the /etc/hosts file (if we could) and forcing the domain to resolve to the internal IP whenever on the internal network.

**Comment by zacwest** [comment]
Allowing the app to use insecure SSL connections (where the hostname and certificate do not match, like above) are logistically difficult and downright impossible in some places -- especially on the watch, where we need to use a background URLSession which doesn't play nicely with this. Please look at the [networking documentation](https://companion.home-assistant.io/docs/troubleshooting/networking) for alternate solutions.
